src/app/features/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.spec.ts

```diff
@@ -99,7 +99,7 @@ describe('HydrateNewVehicleRecordComponent', () => {
       component.handleSubmit();
       actions$.next(createVehicleRecordSuccess);
       tick();
-      expect(navigateSpy).toHaveBeenCalledTimes(1);
+      expect(navigateSpy).toHaveBeenCalledTimes(2);
```

Comment:
This test has been broken. I know this is part of the sprint goal and velocity is needed here, but we cannot submit erroneous code, which becomes the burden of other devs to fix.

`handleSubmit()` is expected to call `router.navigate()` once. But the addition of `this.handleFormState()` into the method made it dependent on the `this.technicalRecordService.editableVehicleTechRecord$` subscription. That is not being mocked, thus line 37 calls `this.navigateBack()`.

If you see and implement my above comment, `expect(navigateSpy).toHaveBeenCalledTimes(1);` would still pass, as it removes the dependency.

Reply:
Thanks for doing it promptly.